foundations/getting/getting.tex

```diff
@@ -0,0 +1,498 @@
+correctly use the command line build tools. Start the version of the Developer
+Command Prompt that matches the architecture for the platform you are targetting. For
+example if you want to build 32-bit OpenSSL then ensure that you start the 32-bit
+Developer Command Prompt. Similarly if you want 64-bit OpenSSL then start the 64-bit
+version.
+
+If you plan to use the MinGW compiler then you should install MSYS2. This includes
+packages for the the MinGW compiler in it \url{http://www.msys2.org}. Make sure you read the
+MSYS2 introduction page at \url{https://github.com/msys2/msys2/wiki/MSYS2-introduction}
+and follow the instructions for setting up MSYS2 with access to the correct compilers.
+After installing MSYS2 with the correct compilers then, if you want to build 32-bit
+OpenSSL, start the MinGW-w64 32-bit Shell or, to build 64-bit OpenSSL, start the
+MinGW-w64 64-bit Shell.
+
+\subsection{Perl} \label{sec:getting-prereq-perl}
+
+Perl is used by multiple scripts within the OpenSSL build system. You will need perl
+version 5.10.0 or later. On Linux you should install this through your package manager.
+You can check the version of Perl that you have by running the following command:
+
+\begin{verbatim}
+$ perl -version
+\end{verbatim}
+
+Be careful to install the correct perl packages. OpenSSL uses a number of core perl
+modules. Full details are in the file NOTES.PERL available in the top level directory
+of the source code archive. On Debian based systems installing the "perl" package
+should give you everything you need. On RedHat based systems install the "perl-core"
+package.
+
+On Windows you should use a version of perl appropriate to your build environment. If
+you are using MSYS2 then you should use the MSYS2 perl package. From the MSYS2 shell
+type:
+
+\begin{verbatim}
+$ pacman -S perl
+\end{verbatim}
+
+If you are using VisualStudio then select a suitable Windows Perl package such as
+ ActiveState's Perl \url{https://www.activestate.com/activeperl} or Strawberry Perl
+\url{http://strawberryperl.com/}. Download the appropriate file from the website and
+run the installer package. Ensure that the Perl executable location is included in
+the \%PATH\% environment variable.
+
+Do not use the wrong perl version for a build environment. For example attempting to
+use ActiveState Perl from MSYS2 or MSYS2 Perl from the VisualStudio developer prompt
+will fail. All of these Windows Perl packages should come with the correct perl modules.
+
+\subsection{Assembler, make and other build tools} \label{sec:getting-prereq-assem}
+
+On Linux you will need a collection of other basic build tools (such as ``GNU as'',
+``ld'', ``ar'' etc). These are distributed as part of the GNU binutils package
+which can be installed as follows for Debian based systems:
+
+\begin{verbatim}
+$ sudo apt-get install binutils
+\end{verbatim}
+
+Or like this on RedHat based systems:
+
+\begin{verbatim}
+$ sudo yum install binutils
+\end{verbatim}
```

Comment:
Doesn't the `gcc` package drag `binutils` along?  I hardly see `gcc` linking anything without having `ld` around in the background, etc...